specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/keys.json

```diff
@@ -0,0 +1,393 @@
+            "description": "Key patch request properties"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Key"
+            }
+          },
+          "default": {
+            "schema": {
+              "$ref": "#/definitions/ErrorContract"
+            },
+            "description": "*** Error Responses: ***\n\n * 500 InternalServerError- Some error occurred while deleting Workspace Key."
+          }
+        },
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "Patch a workspace key": {
+            "$ref": "./examples/UpdateKey.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ErrorDetail": {
+      "description": "Error details",
+      "type": "object",
+      "properties": {
+        "message": {
+          "description": "Error message",
+          "type": "string"
+        },
+        "code": {
+          "description": "Error code",
+          "type": "string"
+        },
+        "target": {
+          "description": "Error target",
+          "type": "string"
+        }
+      }
+    },
+    "ErrorContract": {
+      "title": "Error details.",
+      "description": "Contains details when the response code indicates an error.",
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "The error details.",
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse"
+        }
+      }
+    },
+    "KeyInfoListResult": {
+      "description": "List of keys",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "Link to the next page of results",
+          "type": "string"
+        },
+        "value": {
+          "description": "List of keys",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Key"
+          }
+        }
+      }
+    },
+    "Key": {
+      "description": "A workspace key",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Keys resource properties",
+          "$ref": "#/definitions/KeyProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "KeyPatchInfo": {
+      "description": "KeyInfo patch details",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "description": "Resource tags",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "type": "object"
+        }
+      }
+    },
+    "KeyProperties": {
+      "description": "Key properties",
+      "type": "object",
+      "properties": {
+        "isActiveCMK": {
```

Comment:
why are keys represented as a child resource of the workspace?  If they just contain a key vault url can't that be given to the workspace resource itself during CMK setup?  That is the typical approach

Reply:
We need to keep a list of keys like SQL Server currently supports since our product is advertised as the evolution of SQL DW

Reply:
Please point me to the documentation or swagger specs that describe the SQL API you are mimicing here

Reply:
https://docs.microsoft.com/en-us/rest/api/sql/serverkeys/createorupdate. IsActiveCMK is just our sub-resource signal for CMK with system-assigned managed identity. We don't believe that doing PUT twice on the workspace to enable the CMK flow is a good pattern so we do this pattern only for the provisioning of the resource (in between step1 and step2 of our provisioning)